### PR TITLE
OCaml: add raw odoc generated api files.

### PR DESCRIPTION
**Pull request by recoules**
Publish the api doc generated by `dune build @doc`.

/!\ It is not linked to any previous page of the site.

### CODE REVIEW

File: docs/ocaml/index.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml"><head><title>index (bitwuzla.index)</title><link rel="stylesheet" href="../odoc.css"/><meta charset="utf-8"/><meta name="generator" content="odoc 1.5.2"/><meta name="viewport" content="width=device-width,initial-scale=1.0"/><script src="../highlight.pack.js"></script><script>hljs.initHighlightingOnLoad();</script></head><body><div class="content"><header><nav><a href="../index.html">Up</a> – bitwuzla</nav><h1 id="bitwuzla-index"><a href="#bitwuzla-index" class="anchor"></a>bitwuzla index</h1><nav class="toc"><ul><li><a href="#library-bitwuzla.c">Library bitwuzla.c</a></li><li><a href="#library-bitwuzla.z">Library bitwuzla.z</a></li></ul></nav></header><h2 id="library-bitwuzla.c"><a href="#library-bitwuzla.c" class="anchor"></a>Library bitwuzla.c</h2><p>The entry point of this library is the module: <a href="Bitwuzla_c/index.html"><code>Bitwuzla_c</code></a>.</p><h2 id="library-bitwuzla.z"><a href="#library-bitwuzla.z" class="anchor"></a>Library bitwuzla.z</h2><p>The entry point of this library is the module: <a href="Bitwuzla_z/index.html"><code>Bitwuzla_z</code></a>.</p></div></body></html>
```
Comment (aniemetz) [inline docs/ocaml/index.html]: I would like to have more general documentation on this index page. Ideally, eventually, a quickstart guide and some examples. But at the very least some basic information about the two modules. Should this even be two modules? What is `Bitwuzla_z`?

**Review by aniemetz — COMMENTED** [review]
I would like to tweak the css to be more consistent with the style of the Bitwuzla website, but I can do that after we merge this back. For now, for this PR, it would be great if you can add some basic info on the index page. Also, is there a way to configure some basic things with `odoc`? Like, the header (`bitwuzla` -> `Bitwuzla`), the style and such?

File: docs/ocaml/index.html (hunk 1 above)
Comment (recoules) [reply docs/ocaml/index.html]: I am not sure that `odoc` allows a lot of customizations. I think the idea was to have a consistent "style" across OCaml documentations.

I will try to write some information/guide but I will not have a lot of free time this week so do not expect it before next week-end.

About `Bitwuzla_z`, it is an optional binding allowing to directly extract the value of bitvector to a `Zarith` integer (using `gmp` too). I may change it from optional to required and merge it with `Bitwuzla_c`.
Yet, the end goal is to provide a module `Bitwuzla.Session` that will be a type safe API build on top of `Bitwuzla_c`.

Comment (aniemetz) [reply docs/ocaml/index.html]: Sounds good! No rush, whenever you have time.

**Comment by aniemetz** [comment]
@recoules I committed an initial version of the ocaml docs, already with a tweaked style and place holders in the top `index.html` for quickstart and examples: https://bitwuzla.github.io/docs/ocaml/

Please update your PR to not include the auto-generated files (unless they change, but be careful, you have to manually add https://github.com/bitwuzla/bitwuzla.github.io/blob/master/docs/ocaml/Bitwuzla_c/index.html#L3). Please also do not overwrite the top-level `index.html`, only extend it with some general instructions/documentation for the interface, and in the future, with a quickstart guide and examples.

**Comment by recoules** [comment]
@aniemetz Hi, just a few words to let you know that I am a little bit busy, but I am still experimenting with the API on my free time and I hope to be able to release an update, including some documentation, soon.

Regards,

**Comment by recoules** [comment]
@aniemetz Ok, so I just pushed the new documentation with some examples. Yet, there is an issue with the rendering of code blocks because they are squashed in one big line. Removing the bitwuzla theme solves the issue, but it is not a satisfactory solution.

**Comment by recoules** [comment]
@aniemetz Have you had time to look at it?

**Comment by aniemetz** [comment]
Not yet. I hope I'll find time some time this week.

**Comment by recoules** [comment]
Hi @aniemetz, I updated the index to fix the code display issue.
I think it is still close enough to the style of the website. If you feel it is ok, you can merge it and I will upload the opam package.

File: docs/ocaml/static/css/odoc.css [hunk 2]
```diff
@@ -747,6 +747,26 @@ h1+.modules, h1+.sel {
...
+	font-weight:400;
+	font-style:normal
+}
+.fa,.icon,.rst-content .admonition-title,.rst-content .code-block-caption .headerlink,.rst-content code.download span:first-child,.rst-content dl dt .headerlink,.rst-content h1 .headerlink,.rst-content h2 .headerlink,.rst-content h3 .headerlink,.rst-content h4 .headerlink,.rst-content h5 .headerlink,.rst-content h6 .headerlink,.rst-content p.caption .headerlink,.rst-content table>caption .headerlink,.rst-content tt.download span:first-child,.wy-menu-vertical li.current>a span.toctree-expand,.wy-menu-vertical li.on a span.toctree-expand,.wy-menu-vertical li span.toctree-expand{
```
Comment (aniemetz) [inline docs/ocaml/static/css/odoc.css]:
```suggestion
.icon, .icon:focus {
```
This seems to be sufficient and also fixes the issue with the icon not being found when in focus (when clicking the icon).

**Review by aniemetz — CHANGES_REQUESTED** [review]
Looks great!

Can you change two things:

- Change this:
  `Bitwuzla is an SMT solver for QF_AUFBVFP problems`
  to
  `Bitwuzla is an SMT solver for for the theories of fixed-size bit-vectors, floating-point arithmetic, arrays and uninterpreted functions and their combinations.`
  Bitwuzla has support for BV (quantified bit-vectors) and we are currently in the process of refactoring the quantifier engine to support all combinations of supported theories with quantifiers.
  
- Add a link to `<relative path to>/static/css/custom.css` to all index pages.
  `

**Comment by recoules** [comment]
For memory, here the command to recursively update the style: 
```bash
find docs/ocaml/Bitwuzla/ -type f -exec sed -i -E 's/(([.][.][/])*)[.][.][/]odoc[.]css/\1static\/css\/odoc.css"\/><link rel="stylesheet" href="\1static\/css\/custom.css"\/>/' {} \;
```

**Comment by recoules** [comment]
Hi @aniemetz, I think I made the change you suggested.

**Comment by aniemetz** [comment]
Looks good! Thanks for the update! I'll add a link to the documentation from the main page.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**